src/xc_integrator/local_work_driver/device/scheme1_base.cxx

```diff
@@ -612,47 +537,50 @@ void AoSScheme1Base::eval_kern_exc_vxc_lda( const functional_type& func,
 
   auto base_stack    = data->base_stack;
 
-  
   const bool is_RKS = data->allocated_terms.ks_scheme == RKS;
   const bool is_UKS = data->allocated_terms.ks_scheme == UKS;
+  const bool is_GKS = data->allocated_terms.ks_scheme == GKS;
+  const bool is_2C  = is_UKS or is_GKS;
   const bool is_excgrad = data->allocated_terms.exc_grad;
 
   const size_t npts = data->total_npts_task_batch ;
   
-
-
-  // UKS logic
-  if ( is_UKS ) {
-        // Interleave pos/neg densities before passing it to ExchCXX
-        auto stat = cudaMemcpy2D(base_stack.den_eval_device, 2 * sizeof(double), base_stack.den_pos_eval_device,
-                        1 * sizeof(double), 1 * sizeof(double), npts, cudaMemcpyDeviceToDevice);
-        stat = cudaMemcpy2D(base_stack.den_eval_device + 1, 2 * sizeof(double), base_stack.den_neg_eval_device,
-                        1 * sizeof(double), 1 * sizeof(double), npts, cudaMemcpyDeviceToDevice);
+  auto* dep = base_stack.den_s_eval_device;
+
+  if ( is_2C ) {
+    dep = base_stack.den_eval_device;
+    // Interleave pos/neg densities before passing it to ExchCXX
+    auto  stat = cudaMemcpy2D(base_stack.den_eval_device, 2 * sizeof(double), base_stack.den_s_eval_device,
```

Comment:
I didn't notice this before - but you can't have CUDA specific logic in these files. `memcpy2d` either needs to be abstracted in the device backed or this needs to delegate to a backend-specific function that's separately implemented for CUDA. HIP and SYCL (you don't need to worry about the latter two for now)